### PR TITLE
feat(hudi): add apache hudi writer

### PR DESCRIPTION
**Pull request by lyogev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -340,5 +340,60 @@ catalog:
 
 Check out the [examples](examples/hive) and the [E2E test](e2e/hive)
 
+
+#### Apache Hudi
+Metorikku supports reading/writing with [Apache Hudi](https://github.com/apache/incubator-hudi).
+
+Hudi is a very exciting project that basically allows upserts and deletes directly on top of the data lake.
```
Comment (ofirventura) [inline README.md]: call it parquet files or something like that - maybe they dont have a lake

File: config/job_config_sample.yaml [hunk 2]
```diff
@@ -95,6 +95,26 @@ output:
...
+    # This controls the level of parallelism of hudi writing (should be similar to shuffle partitions)
+    parallelism: 1
+    # upsert/insert
+    operation: upsert
+    # COPY_ON_WRITE/MERGE_ON_READ
+    storageType: COPY_ON_WRITE
+    # Maximum number of versions to retain
+    maxVersions: 1
+    # Hive database to use when writing
+    hiveDB: default
```
Comment (ofirventura) [inline config/job_config_sample.yaml]: what other values can be here? is this a url?

File: README.md (hunk 1 above)
Comment (ofirventura) [inline README.md]: maybe call it parquet not data lake

File: README.md [hunk 3]
```diff
@@ -340,5 +340,60 @@ catalog:
...
+```
+
+#### Metric config
+```yaml
+dataFrameName: test
+  outputType: Hudi
+  outputOptions:
+    path: test.parquet
+    # The key to use for upserts
+    keyColumn: userkey
```
Comment (ofirventura) [inline README.md]: compound key ?

File: config/metric_config_sample.yaml [hunk 4]
```diff
@@ -144,3 +144,23 @@ output:
       # you can add below any option supported by "Elasticsearch for Apache Hadoop"
       extraOptions:
         es.index.auto.create: false
+  - outputType: Hudi
+    outputOptions:
```
Comment (ofirventura) [inline config/metric_config_sample.yaml]: whats mandatory and whats optional ? maybe clear the explanation for the delete part ?

File: docker/hive/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	&& tar -xzf apache-hive-$HIVE_VERSION-bin.tar.gz \
+	&& mv apache-hive-$HIVE_VERSION-bin $HIVE_HOME \
+	&& rm apache-hive-$HIVE_VERSION-bin.tar.gz
+
+ENV MYSQL_CONNECTOR_VERSION=5.1.47
+RUN wget -q http://central.maven.org/maven2/mysql/mysql-connector-java/$MYSQL_CONNECTOR_VERSION/mysql-connector-java-$MYSQL_CONNECTOR_VERSION.jar \
+    && mv mysql-connector-java-$MYSQL_CONNECTOR_VERSION.jar $HIVE_HOME/lib
+
+RUN apt-get update && apt-get install -y ant
+RUN wget -q http://central.maven.org/maven2/com/uber/hoodie/hoodie-hive-bundle/0.4.5/hoodie-hive-bundle-0.4.5.jar \
```
Comment (ofirventura) [inline docker/hive/Dockerfile]: is this version gonna change in the future when we upgrade spark etc ? if so, please extract it to variable, you can put it with HADOOP_VERSION in the compose if you want

File: config/job_config_sample.yaml [hunk 6]
```diff
@@ -95,6 +95,26 @@ output:
     user: username
     password: password
     driver: "org.postgresql.Driver"
+  # Apache Hudi
+  hudi:
+    dir: /path/to/parquet/output
+    # This controls the level of parallelism of hudi writing (should be similar to shuffle partitions)
+    parallelism: 1
+    # upsert/insert
```
Comment (ofirventura) [inline config/job_config_sample.yaml]: and bulkinsert?

File: src/main/scala/com/yotpo/metorikku/configuration/job/output/Hudi.scala [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.yotpo.metorikku.configuration.job.output
+
+case class Hudi(dir: String,
+                parallelism: Option[String],
+                maxFileSize: Option[String],
+                operation: Option[String],
```
Comment (ofirventura) [inline src/main/scala/com/yotpo/metorikku/configuration/job/output/Hudi.scala]: what are the default values for all this variables if I define only the dir ? copy on write, upsert, etc.... maybe add something about this on the readme

File: src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala [hunk 8]
```diff
@@ -0,0 +1,170 @@
+package com.yotpo.metorikku.output.writers.file
+
+import java.util.Optional
+
+import com.yotpo.metorikku.configuration.job.output.Hudi
+import com.yotpo.metorikku.output.Writer
+import org.apache.spark.sql.{DataFrame, DataFrameWriter, SaveMode}
+
+// REQUIRED: -Dspark.serializer=org.apache.spark.serializer.KryoSerializer
```
Comment (ofirventura) [inline src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala]: add serializer requirements to readme too ?

File: src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+      case (Some(path), Some(output)) => Option(output.dir + "/" + path)
+      case (Some(path), None) => Option(path)
+      case _ => None
+    }
+    path match {
+      case Some(filePath) => writer.option("path", filePath)
+      case None =>
+    }
+
+    // Mandatory
```
Comment (ofirventura) [inline src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala]: its not really clear whats mandatory and whats not

File: README.md (hunk 3 above)
Comment (lyogev) [reply README.md]: It's called key column by hudi, I tried to keep it similar, but it can be compound of course

Comment (ofirventura) [reply README.md]: how can I separate it ? can I write for example: id,name,email ?

**Pull request merged**